src/java/org/apache/cassandra/service/consensus/TransactionalMode.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.consensus;
+
+import org.apache.cassandra.db.ConsistencyLevel;
+import org.apache.cassandra.service.accord.IAccordService;
+
+/*
+ * Configure the transactional behavior of a table. Enables accord on a table and defines how it mixes with non-serial writes
+ *
+ * For Accord transactions to function correctly when mixed with non-SERIAL writes it's necessary for the writes to occur through Accord.
+ *
+ * Accord will also use this configuration to determine what consistency level to perform its reads
+ * at since it will need to be able to read data written at non-SERIAL consistency levels.
+ *
+ * BlockingReadRepair will also use this configuration to determine how BRR mutations are applied. For migration
+ * and accord the BRR mutations will be applied as Accord transactions so that BRR doesn't expose Accord to
+ * uncommitted Accord data that is being RRed. This can occur when Accord has applied a transaction at some, but not
+ * all replica since Accord defaults to asynchronous commit.
+ *
+ * By routing repairs through Accord it is guaranteed that the Accord derived contents of the repair have already been applied at any
+ * replica where Accord applies the transaction. This also prevents BRR from breaking atomicity of Accord writes.
+ *
+ * If they are not written through Accord then reads through Accord will be required to occur at
+ * consistency level compatible with the non-serial writes preventing single replica reads from being performed
+ * by Accord. It will also require Accord to perform read repair of non-serial writes.
+ *
+ * Even then there is the potential for Accord to inconsistently execute transactions at different replicas
+ * because different coordinators for an Accord transaction may encounter different non-SERIAL write state and
+ * race to commit different outcomes for the transaction.
+ *
+ * This is different from Paxos because Paxos performs consensus on the actual values to be applied so recovery
+ * coordinators will always produce a consistent state when applying a transaction. Accord performs consensus on
+ * the execution order of transaction and different coordinators witnessing different states not managed by Accord
+ * can produce multiple outcomes for a transaction.
+ *
+ * // TODO to safely migrate you would have to route all writes through Accord with the current implementation
+ * // We could do it by range instead in the migration version, but then we need to know when all in flight writes
+ * // are done before marking a range as migrated. Would waiting out the timeout be enough (timeout bugs!)?
+ */
+public enum TransactionalMode
```

Comment:
Do you mean the name TransactionalMode? How is it problematic? 

If you mean the individual setting names, I'd agree they could be improved a bit. `mixed` could be called `mixed_unsafe`. I'm not sure what `migration` should be called.

It's not clear how reads are handled for each of these in either case.

Do you have a suggestion for how interop should be broken out? Do you mean as a 3rd schema option?

It sounds like we will want configurable defaults for on/off and/or interop mode that are applied in CreateTableStatement

Reply:
The name `TransactionalMode` is problematic because this is really hyper focused on interoperability with Accord and we have `LWT` so the word `Transactional` makes it seem like this is how `LWT`s are configured and it's not just that. It's really `AccordInteroperabilityMode`.

I was thinking it might be cleaner to just have an on/off for interoperability, but I have warmed up to the current approach. I just think maybe allowing changing between different interoperability modes should be allowed. More in my comment on alter table.

Reply:
> It's really AccordInteroperabilityMode.

Its main function is turning accord on or off for a table. Interop modes are included here to avoid having a separate config setting that only makes sense if this one is on

Regarding the naming of the name and/or it's settings, do you have any suggestions?

Reply:
for mode names, how about:

Reply:
Regarding the name TransactionalMode, I see the names Transactional and Accord as having different functions. One refers to the C* feature and and the other refers to the consensus protocol, respectively. Since configuration is user facing, I don’t think naming it accord is appropriate (and think AccordSpec should be renamed to TransactionalSpec). This is similar to how we have a user feature call lwts, and they are implemented w/ paxos.

Reply:
OK, that makes sense.